workflows/eager.nf

```diff
@@ -26,19 +26,52 @@ if ( params.metagenomics_complexity_tool == 'prinseq' && params.metagenomics_pri
             exit 1, ("[nf-core/eager] ERROR: Metagenomics: You picked PRINSEQ++ with 'dust' mode but provided an entropy score. Please specify a dust filter threshold using the --metagenomics_prinseq_dustscore flag")
     }
 }
-if ( params.metagenomics_complexity_tool == 'prinseq' && params.metagenomics_prinseq_mode == 'entropy' && params.metagenomics_prinseq_dustscore != 0.5 ) {
-    // dust score was set but entropy method picked. If no entropy-score provided, assume it was an error and fail
-    if (params.metagenomics_complexity_entropy == 0.3) {
-            exit 1, ("[nf-core/eager] ERROR: Metagenomics: You picked PRINSEQ++ with 'entropy' mode but provided a dust score. Please specify an entropy filter threshold using the --metagenomics_complexity_entropy flag")
+
+// metagenomics related parameter checks
+if ( params.run_metagenomics ) {
+    // failing metagenomics parameter combinations/declarations
+    if ( params.metagenomics_complexity_tool == 'prinseq' && params.metagenomics_prinseq_mode == 'entropy' && params.metagenomics_prinseq_dustscore != 0.5 ) {
+        // dust score was set but entropy method picked. If no entropy-score provided, assume it was an error and fail
+        if (params.metagenomics_complexity_entropy == 0.3) {
+                exit 1, ("[nf-core/eager] ERROR: Metagenomics: You picked PRINSEQ++ with 'entropy' mode but provided a dust score. Please specify an entropy filter threshold using the --metagenomics_complexity_entropy flag")
```

Comment:
replace all `exit` with `error()` (something to do with we want Nextflow to handle this rather than pure groovy... see taxprofiler for an example)